hack/ci/e2e-conformance.sh

```diff
@@ -25,47 +25,143 @@ CLUSTER_NAME=${CLUSTER_NAME:-"test1"}
 
 REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
 
-# our exit handler (trap)
-cleanup() {
+# dump logs from kind and all the nodes
+dump-logs() {
   # always attempt to dump logs
   kind "export" logs --name="clusterapi" "${ARTIFACTS}/logs" || true
+
+  # iterate through any nodes we brought up and collect logs
+  gcloud compute instances list --project "${GCP_PROJECT}" --format='value(name, zone)' \
+  | grep "${CLUSTER_NAME}" | while read node_name node_zone; do
+    echo "collecting logs from ${node_name}"
+    dir="${ARTIFACTS}/logs/${node_name}"
+    mkdir -p ${dir}
+
+    gcloud compute instances get-serial-port-output --project "${GCP_PROJECT}" \
+      --zone "${node_zone}" --port 1 "${node_name}" > "${dir}/serial-1.log" || true
+
+    ssh-to-node "${node_name}" "${node_zone}" "sudo chmod -R a+r /var/log" || true
+    gcloud compute scp --recurse --project "${GCP_PROJECT}" --zone "${node_zone}" \
+      "${node_name}:/var/log/cloud-init.log" "${node_name}:/var/log/cloud-init-output.log" "${dir}" || true
+
+    ssh-to-node "${node_name}" "${node_zone}" "sudo journalctl --output=short-precise" > "${dir}/systemd.log" || true
+  done
+
+  timeout 120 gcloud logging read --order=asc \
+    --freshness="3h" \
+    --format='table(timestamp,jsonPayload.resource.name,jsonPayload.event_subtype)' \
+    --project "${GCP_PROJECT}" > "${ARTIFACTS}/logs/activity.log" || true
+}
+
+# our exit handler (trap)
+cleanup() {
+  # dump all the logs
+  dump-logs
+
   # KIND_IS_UP is true once we: kind create
   if [[ "${KIND_IS_UP:-}" = true ]]; then
-    kubectl \
+    timeout 60 kubectl \
       --kubeconfig=$(kind get kubeconfig-path --name="clusterapi") \
       delete cluster test1 || true
-    kubectl \
+     timeout 60 kubectl \
```

Comment:
60 seconds might be a bit too short of a timeout here, maybe give it at least 10-15 minutes here?

Reply:
Ack, i'll bump it up to 10 and we can tweak it